include/seqan3/core/copyable_wrapper.hpp

```diff
@@ -0,0 +1,248 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::copyable_wrapper.
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <functional>
+#include <optional>
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3
+{
+
+/*!\brief Helper concept for seqan3::copyable_wrapper.
+ * \ingroup core
+ * \see https://en.cppreference.com/w/cpp/ranges/copyable_wrapper
+ * \noapi{Exposition only.}
```

Comment:
why can't this be in `seqan3::detail`?

Reply:
I did not because `semiregular_box` wasn't...but because it's `core`, we can just move it to detail, right?

`Exposition only.` only refers to the standard, where it is considered as such. Won't matter if we move it to detail

Reply:
I missed that in my former re-review.

IMHO, I would avoid "exposition only" mentions since we already have the detail namespace and the \noapi tag for stuff that we don't give guarantees for. Because this doesn't seem to be something we will expose to the user at any given point, I see no reason to NOT make it detail?